js/iframe-coordinator/libs/BackgroundClient.ts

```diff
@@ -0,0 +1,35 @@
+import { ToastingClient, ToastOptions } from './types';
+import {
+  WORKER_MESSAGING_PROTOCOL_NAME,
+  WorkerToHostMessageTypes
+} from './workers/constants';
+
+const ctx: Worker = self as any;
+
+export default class BackgroundClient implements ToastingClient {
+  public requestToast(
+    message: string,
+    { title = null, custom = null }: ToastOptions = {}
+  ): void {
+    this._publishMessageToHost(WorkerToHostMessageTypes.ToastRequest, {
+      title,
+      message,
+      custom
+    });
+  }
+
+  // TODO Change this to URL based?
+  public requestNavigation(route: string): void {
+    this._publishMessageToHost(WorkerToHostMessageTypes.NavRequest, {
+      fragment: route
+    });
+  }
+
+  private _publishMessageToHost(msgType: string, msg?: object): void {
+    ctx.postMessage({
+      protocol: WORKER_MESSAGING_PROTOCOL_NAME,
```

Comment:
Having worked with a lot of messaging between iframes, we found that having a protocol field with a specific string is a great way to prune out invalid or unintentional messages.  Other devs are not always careful to domain-restrict or target their messages appropriately.

I also usually add an agent name and version so we can track api clients, but that can come later.  I think we should adapt this pattern to all of our messages

Reply:
how is this different than the msgType?  It's really just acting like a combined key with the msg type.  I guess if you have had issues before, i'll defer.  Just seems extra cautious.

Reply:
The issue is that a lot of sites are using some form of iframe/postMessage these days and a growing number of pages and frameworks are using workers.  And, I've seen a fair share that don't do a good job targeting their messages to a particular frame or adding a targetOrigin.  

Having an unambiguous field like protocol and giving it a unique value allows us to effectively filter in our code.  In other projects, I had a ton of false-positive error logging because I was processing messages (and logging warnings of malformed payloads) from embedded urls that were pushing events over postMessage.  They all use generic fields like type, msgType, etc.

Now, we could get a similar result by namespacing our msgType values.  Something like `com.genesys:iframe-coordinator:requestToast` would allow us to filter effectively and also reduces the likelihood that our messages would be matched and consumed by others.

So, I could certainly go this way as well if we'd rather not add an extra field.

Reply:
I think i prefer a separate field so the msgType becomes more meaningful/easier to work with.